File: java-engine/src/main/java/hu/elteik/knowledgelab/javaengine/core/models/Graph.java
package hu.elteik.knowledgelab.javaengine.core.models;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class Graph<NodeType extends Node, EdgeType extends Edge> {

    private List<NodeType> nodeList;
    private List<EdgeType> edgeList;

}

File: frontend/src/app/models/core/Graph.ts
import { Node } from "./Node";
import { Edge } from "./Edge";

export class Graph<NodeType extends Node, EdgeType extends Edge> {
  nodeList: NodeType[];
  edgeList: EdgeType[];

  constructor(nodeList: NodeType[], edgeList: EdgeType[]) {
    this.nodeList = nodeList;
    this.edgeList = edgeList;
  }
}
